src/shogun/modelselection/NewGridSearch.cpp

```diff
@@ -0,0 +1,891 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Gil Hoben
+ */
+
+#include <shogun/distance/Distance.h>
+#include <shogun/evaluation/CrossValidation.h>
+#include <shogun/kernel/Kernel.h>
+#include <shogun/modelselection/NewGridSearch.h>
+#include <shogun/util/factory.h>
+
+using namespace shogun;
+
+const std::string ParameterNode::delimiter = "::";
+
+ParameterNode::ParameterNode(CSGObject* model)
+{
+	SG_REF(model)
+	m_parent = model;
+	SG_SDEBUG("%s refcount: %d\n", model->get_name(), m_parent->ref_count())
```

Comment:
FYI if you set the debug level to the lowest, you will see whenever ref-counts are changed

Reply:
Yea, true, it just had this as SG_SPRINT, and then replaced it with DEBUG in the end